go/vt/vtctl/reparentutil/reparent_sorter.go

```diff
@@ -26,68 +26,69 @@ import (
+type reparentSorter struct {
 	tablets   []*topodatapb.Tablet
 	positions []mysql.Position
 }
 
-// NewERSIntermediateSourceSorter creates a new ERSIntermediateSourceSorter
-func NewERSIntermediateSourceSorter(tablets []*topodatapb.Tablet, positions []mysql.Position) *ERSIntermediateSourceSorter {
-	return &ERSIntermediateSourceSorter{
+// newReparentSorter creates a new reparentSorter
+func newReparentSorter(tablets []*topodatapb.Tablet, positions []mysql.Position) *reparentSorter {
+	return &reparentSorter{
 		tablets:   tablets,
 		positions: positions,
 	}
 }
 
 // Len implements the Interface for sorting
-func (ersISSorter *ERSIntermediateSourceSorter) Len() int { return len(ersISSorter.tablets) }
+func (rs *reparentSorter) Len() int { return len(rs.tablets) }
 
 // Swap implements the Interface for sorting
-func (ersISSorter *ERSIntermediateSourceSorter) Swap(i, j int) {
-	ersISSorter.tablets[i], ersISSorter.tablets[j] = ersISSorter.tablets[j], ersISSorter.tablets[i]
-	ersISSorter.positions[i], ersISSorter.positions[j] = ersISSorter.positions[j], ersISSorter.positions[i]
+func (rs *reparentSorter) Swap(i, j int) {
+	rs.tablets[i], rs.tablets[j] = rs.tablets[j], rs.tablets[i]
+	rs.positions[i], rs.positions[j] = rs.positions[j], rs.positions[i]
 }
 
 // Less implements the Interface for sorting
-func (ersISSorter *ERSIntermediateSourceSorter) Less(i, j int) bool {
+func (rs *reparentSorter) Less(i, j int) bool {
 	// Returning "true" in this function means [i] is before [j] in the sorting order,
 	// which will lead to [i] be a better candidate for promotion
 
 	// Should not happen
 	// fail-safe code
-	if ersISSorter.tablets[i] == nil {
+	if rs.tablets[i] == nil {
 		return false
 	}
-	if ersISSorter.tablets[j] == nil {
+	if rs.tablets[j] == nil {
 		return true
 	}
 
-	if !ersISSorter.positions[i].AtLeast(ersISSorter.positions[j]) {
+	if !rs.positions[i].AtLeast(rs.positions[j]) {
 		// [i] does not have all GTIDs that [j] does
 		return false
 	}
-	if !ersISSorter.positions[j].AtLeast(ersISSorter.positions[i]) {
+	if !rs.positions[j].AtLeast(rs.positions[i]) {
 		// [j] does not have all GTIDs that [i] does
 		return true
 	}
 
 	// at this point, both have the same GTIDs
 	// so we check their promotion rules
-	jPromotionRule := PromotionRule(ersISSorter.tablets[j])
-	iPromotionRule := PromotionRule(ersISSorter.tablets[i])
+	jPromotionRule := PromotionRule(rs.tablets[j])
+	iPromotionRule := PromotionRule(rs.tablets[i])
 	return !jPromotionRule.BetterThan(iPromotionRule)
 }
 
-// sortTabletsForERS sorts the tablets, given their positions for emergency reparent shard
-func sortTabletsForERS(tablets []*topodatapb.Tablet, positions []mysql.Position) error {
+// sortTabletsForReparent sorts the tablets, given their positions for emergency reparent shard and planned reparent shard
+// primary sorting is via the tablet positions, and ties are broken by the promotion rules.
```

Comment:
nit: this is an unfortunate name collision, because you mean "primary" as in "the main sorting" and not the tablet type.

suggestion:

> Tablets are sorted first by their replication positions, with ties broken by the promotion rules.

(also can you add some punctuation to the previous line? i had to read it a few times to realize this wasn't all one sentence)

Reply:
Yes, that's an unfortunate conflict 😅. I didn't realise it when I wrote it. Thankyou for pointing it out. I have taken your suggestion